picard/ui/options/acousticbrainz.py

```diff
@@ -0,0 +1,147 @@
+from PyQt5 import (
+    QtCore,
+    QtWidgets,
+)
+
+from picard.acousticbrainz import (
+    ab_setup_extractor,
+    get_extractor,
+)
+from picard.config import (
+    BoolOption,
+    TextOption,
+    get_config,
+)
+from picard.util import webbrowser2
+
+from picard.ui.options import (
+    OptionsCheckError,
+    OptionsPage,
+    register_options_page,
+)
+from picard.ui.ui_options_acousticbrainz import Ui_AcousticBrainzOptionsPage
+
+
+class AcousticBrainzOptionsPage(OptionsPage):
+
+    NAME = "AcousticBrainz"
+    TITLE = N_("AcousticBrainz")
+    PARENT = None
+    SORT_ORDER = 45
+    ACTIVE = True
+    HELP_URL = '/config/options_acousticbrainz.html'
+
+    options = [
+        BoolOption("setting", "use_acousticbrainz", False),
+        TextOption("setting", "acousticbrainz_extractor", ""),
+        TextOption("setting", "acousticbrainz_extractor_version", ""),
+        TextOption("setting", "acousticbrainz_extractor_sha", ""),
+    ]
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self._extractor_valid = True
+        self.ui = Ui_AcousticBrainzOptionsPage()
+        self.ui.setupUi(self)
+        self.ui.disable_acoustic_features.clicked.connect(self.update_groupboxes)
+        self.ui.use_acoustic_features.clicked.connect(self.update_groupboxes)
+        self.ui.acousticbrainz_extractor.textChanged.connect(self._acousticbrainz_extractor_check)
+        self.ui.acousticbrainz_extractor_browse.clicked.connect(self.acousticbrainz_extractor_browse)
+        self.ui.acousticbrainz_extractor_download.clicked.connect(self.acousticbrainz_extractor_download)
+        self._acousticbrainz_extractor_info = ""
+        self._tagger = QtCore.QCoreApplication.instance()
+        self._config = get_config()
+
+    def load(self):
+        if self._config.setting["use_acousticbrainz"]:
+            self.ui.use_acoustic_features.setChecked(True)
+        else:
+            self.ui.disable_acoustic_features.setChecked(True)
+        self.ui.acousticbrainz_extractor.setPlaceholderText("streaming_extractor_music")
+        self.ui.acousticbrainz_extractor.setText(self._config.setting["acousticbrainz_extractor"])
+        self.update_groupboxes()
+
+    def save(self):
+        if self.ui.use_acoustic_features.isChecked():
+            self._config.setting["use_acousticbrainz"] = True
+        else:
+            self._config.setting["use_acousticbrainz"] = False
+        self._config.setting["acousticbrainz_extractor"] = self.ui.acousticbrainz_extractor.text()
+
+    def update_groupboxes(self):
+        if self.ui.use_acoustic_features.isChecked():
+            self.ui.acousticbrainz_settings.setEnabled(True)
+            # If current path to extractor isn't valid, look for an extractor
+            if not os.path.exists(self.ui.acousticbrainz_extractor.text()):
+                self.ui.acousticbrainz_extractor.setText(get_extractor())
+        else:
+            self.ui.acousticbrainz_settings.setEnabled(False)
```

Comment:
It could be shorten a bit:

```python
        enabled = self.ui.use_acoustic_features.isChecked()
        self.ui.acousticbrainz_settings.setEnabled(enabled)
        if enabled and not os.path.exists(self.ui.acousticbrainz_extractor.text()):
            # If current path to extractor isn't valid, look for an extractor
            self.ui.acousticbrainz_extractor.setText(get_extractor())
        self._acousticbrainz_extractor_check()
```